addons/calendar/models/calendar_recurrence.py

```diff
@@ -604,3 +604,18 @@ def _get_rrule(self, dtstart=None):
             freq_to_rrule(freq), **rrule_params
         )
 
+    def _is_event_over(self):
+        """Check if all events in this recurrence are in the past.
+        :return: True if all events are over, False otherwise
+        """
+        self.ensure_one()
+        if not self.calendar_event_ids:
+            return False
+
+        now = fields.Datetime.now()
+        today = fields.Date.today()
+
+        return all(
+            (event.stop_date < today if event.allday else event.stop < now)
+            for event in self.calendar_event_ids
+        )
```

Comment:
we don't have to check if `stop_date` and `stop` are set? 🤔 

I saw we check first if `stop_date` and `stop` are not falsy in the same method in calendar.event model. 🤷🏼‍♂️ 

Actually, why not doing that:

```suggestion
    def _is_event_over(self):
        """Check if all events in this recurrence are in the past.
        :return: True if all events are over, False otherwise
        """
        self.ensure_one()
        if not self.calendar_event_ids:
            return False

        return all(
            event._is_event_over()
            for event in self.calendar_event_ids
        )
```